### PR TITLE
Moves to jemalloc 

### PR DESCRIPTION
**Pull request by ashfurrow**
Moves to Sir Boops' Docker setup for jemalloc. 

I had a very difficult time testing this locally because it required HTTPS – Mastodon's docs specify not to try running Docker locally so we'll have to test it in prod 🙃

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,88 +1,110 @@
...
+# Taken from Sir Boops' Dockerfile: https://git.sergal.org/Sir-Boops/docker-mastodon/src/branch/master/Dockerfile
+# For the purposes of using jemalloc.
 
-LABEL maintainer="https://github.com/tootsuite/mastodon" \
-      description="Your self-hosted, globally interconnected microblogging community"
+FROM ubuntu:18.04 as build-dep
 
-ARG UID=991
-ARG GID=991
+EXPOSE 3000 4000
```
Comment (ashfurrow) [inline Dockerfile]: This is my only addition from Sir Boops' Dockerfile, linked to above. Not sure if it's necessary, I'm going to ping them and ask.

**Comment by Sir-Boops** [comment]
> so we'll have to test it in prod

Masto makes things so easy to test doesn't it! /s

Might want to only test a little piece of it first...like only switching sidekiq over then if that's all good the rest can get moved! :+1:

**Review by Sir-Boops — COMMENTED** [review]
This should work fine but just be aware

`/mastodon` in this container is just a link `/opt/mastodon`

**Comment by ashfurrow** [comment]
That's a good idea! The services all _booted_ locally, I just couldn't access them 😅 I have half a mind to spin up a VM somewhere, give it a DNS entry and a lets encrypt cert, and test it there first. We'll seeeeeeeee 😸

**Comment by ashfurrow** [comment]
> `/mastodon` in this container is just a link `/opt/mastodon`

I did notice these were different from what I had before, do you mean "link to `/opt/mastodon`" on the host VM? Thanks again for answering all my questions 😅

**Review by Sir-Boops — CHANGES_REQUESTED** [review]
The full path to masto inside the container is
`/opt/mastodon` not the `/mastodon` in the official.

My container DOES link `/mastodon` to `/opt/mastodon`

So these *might* work fine but changing the internal mounts to `/opt/mastodon/public/packs` Might also be needed!

**Comment by ashfurrow** [comment]
Ah, gotcha! Yup I'll need to change the mounts in docker-compose.yml to point to the full `/opt/mastodon/public/...`.

It looks like [this `COPY` line](https://github.com/ashfurrow/mastodon/pull/8/files#diff-3254677a7917c6c01f55212f86c57fbfR87) might need a change too, but I'll look into what `--from-build-deps` means.

**Comment by ashfurrow** [comment]
Oh, okay. Hmm – since my code is hosted in `/home/ash/mastodon` (I know, not the best 😬) I'll likely have to update that `COPY` instruction.

**Comment by Sir-Boops** [comment]
is `/home/ash/mastodon` a built version of masto or just the source?

Like local gems installed yarn packages installed to it

**Comment by ashfurrow** [comment]
Oh! It's just the source.

**Comment by Sir-Boops** [comment]
Do you have any custom things in the source or no?

If ya don't then you don't have to use that one :p 

If you do or just want to use something like this ( Haven't tested this but should work for ya! ) -> https://gist.github.com/Sir-Boops/d748a5be6da4f02b41ea8b0f54f9c62e

**Comment by ashfurrow** [comment]
Cool thanks! I do have some customizations. I appreciate the gist!

**Comment by ashfurrow** [comment]
SIR BOOPS, IT WORKED

screen shot 2018-08-22 at 9 19 31 pm

Thanks again for everything! I've blocked off some time on Saturday to get this from staging onto the actual m.t instance. I'll let you know how it goes 👍

**Comment by Sir-Boops** [comment]
Yay! Thanks for helping me find those few bugs with it!

And have fun pushing to prod~

**Comment by ashfurrow** [comment]
Okay I'm happy with this, going to merge. Memory use is still stabilizing and I, uh, committed the sin of forgetting to benchmark memory use in detail before changing something designed to help with memory problems 🙈 I feel a lot more comfortable with these tools and I learned a lot – thanks again for your help!

**Pull request merged**